### PR TITLE
ROC-4783:Update court calculations - claimant immediately scenario

### PR DESCRIPTION
**Pull request by varma02**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-4783

### Change description
Update court calculations - claimant immediately scenario

### Work checklist

- [x] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/features/claimant-response/routes/payment-option.ts [hunk 1]
```diff
@@ -61,12 +61,17 @@ export class PaymentOptionPage extends AbstractPaymentOptionPage<DraftClaimantRe
...
-      if (claimResponse.paymentIntention.paymentOption === PaymentOption.BY_SPECIFIED_DATE) {
-        courtOfferedPaymentIntention.paymentDate = paymentPlanFromDefendantFinancialStatement.calculateLastPaymentDate()
-        courtOfferedPaymentIntention.paymentOption = PaymentOption.BY_SPECIFIED_DATE
+      } else {
+        const paymentPlanConvertedToMonthlyFrequency: PaymentPlan = paymentPlanFromDefendantFinancialStatement.convertTo(Frequency.MONTHLY)
 
+        courtOfferedPaymentIntention.paymentOption = PaymentOption.INSTALMENTS
+        courtOfferedPaymentIntention.repaymentPlan = {
+          firstPaymentDate: paymentPlanConvertedToMonthlyFrequency.startDate,
+          instalmentAmount: Math.round(paymentPlanConvertedToMonthlyFrequency.instalmentAmount * 100) / 100,
```
Comment (robertparkinson) [inline src/main/features/claimant-response/routes/payment-option.ts]: the `instalmentAmount * 100) / 100` is used in 3 places in this file. can you please pull it out to a function now and add some tests for it?

Comment (varma02) [reply src/main/features/claimant-response/routes/payment-option.ts]: i have updated it to use lodash round function. Don't think we need test now!

if you agree, i will create seperate PR to update to lodash round function in other court determination calculations...

File: src/main/features/claimant-response/routes/payment-option.ts [hunk 2]
```diff
@@ -55,18 +56,23 @@ export class PaymentOptionPage extends AbstractPaymentOptionPage<DraftClaimantRe
...
+        courtOfferedPaymentIntention.repaymentPlan = {
+          firstPaymentDate: paymentPlanConvertedToMonthlyFrequency.startDate,
+          instalmentAmount: _.round(paymentPlanConvertedToMonthlyFrequency.instalmentAmount,2),
+          paymentSchedule: Frequency.toPaymentSchedule(paymentPlanConvertedToMonthlyFrequency.frequency),
+          completionDate: paymentPlanConvertedToMonthlyFrequency.calculateLastPaymentDate(),
+          paymentLength: paymentPlanConvertedToMonthlyFrequency.calculatePaymentLength()
+        }
         return courtOfferedPaymentIntention
       }
       return undefined
```
Comment (andrewwa-kainos) [inline src/main/features/claimant-response/routes/payment-option.ts]: this line is now unreachable

File: src/main/features/claimant-response/routes/payment-option.ts [hunk 3]
```diff
@@ -160,7 +167,7 @@ export class PaymentOptionPage extends AbstractPaymentOptionPage<DraftClaimantRe
       response.statementOfMeans,
       response.defendant.type,
       PaymentOptionPage.getDateOfBirth(response.defendant))
-    return disposableIncome === 0 ? 0 : Math.round(disposableIncome * 100) / 100
+    return disposableIncome === 0 ? 0 : _.round(disposableIncome,2)
```
Comment (andrewwa-kainos) [inline src/main/features/claimant-response/routes/payment-option.ts]: What is `_.round(0, 2)`? The conditional might be unnecessary

**Pull request merged**